calcom/cal.com
# feat: automatic no show

## What does this PR do?


- Fixes https://github.com/calcom/cal.com/issues/16689 (GitHub issue number)
- Fixes CAL-4357 (Linear issue number - should be visible at the bottom of the GitHub issue description)

![Screenshot 2024-09-24 at 4 24 00 PM](https://github.com/user-attachments/assets/fc5611f5-a7c6-4ec1-80f5-b8d5286247ad)


- Hosts need to be logged in to cal.com when they join cal video meeting so that we can detect whether they joined the meeting or not.




## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

1) Go to any event type webhooks page
2) Create webhook with two trigger events "After hosts doesn't join cal vide" , "After guest doesn't join cal video" and select time as 5 minutes.
3) Create a booking and join the cal video link (make sure you are logged in as host)
4) then you can run yarn dev:cron and update the scheduledAt date(in prisma studio) to any past date to execute the cron job tasks.

~3) you can test the task `triggerNoShowHost` by importing this function in~ ~packages/features/bookings/lib/handleNewBooking/scheduleNoShowTriggers.ts and~

```
~triggerHostNoShow(JSON.stringify({ bookingId:booking.id, webhook: subscribersHostsNoShowStarted[0])}~
```

~and try booking a meeting.~ 

~3) Make a POST request to http://localhost:3000/api/cron/calVideoNoShowWebhookTriggers to trigger the webhooks.
You can comment this code~

~and change the startAfter date to past date like 26th sept so that these jobs are executed when you make the POST request~
